app/src/main/java/org/systers/mentorship/view/activities/SignUpActivity.kt

```diff
@@ -35,9 +36,22 @@ class SignUpActivity : BaseActivity() {
     private var needsMentoring: Boolean = false
     private var isAvailableForBoth: Boolean = false
 
+    // checking if any of the fields is empty at an interval of every 1.5 seconds
+    private val handler = Handler()
+    private val checkInput = object : Runnable{
+        override fun run() {
+            getDetails()
+
+            btnSignUp.isEnabled = validateDetails(false)
+
+            handler.postDelayed(this, 1500)
```

Comment:
Why do you need this specific delay? I don't understand 🤔

Reply:
> Why do you need this specific delay? I don't understand 🤔

this will allow to check the empty fields at an interval of every 1.5 seconds, otherwise the handler would be constantly running till there are no empty fields which might slow the app, once there are no empty fields the signup button will become enabled

Reply:
the delay is just to prevent any unnecessary slowing down of the app, that's what my thought was :-)

Reply:
@isabelcosta any further changes required ?

Reply:
Can you use an on text change listener instead?

Reply:
hmm lemme try it..